### DISCUSSION TITLE
save_charts_path doesn't work?

### DISCUSSION THREAD

**Discussion by notmanas** [sinaptik-ai/pandas-ai#507] [Q&A]
Hi,
Here is the code that I'm using -

```
user_defined_path = "tmp_charts"

df = SmartDataframe("patients.csv",
                    config={"enable_cache": False, "llm": llm,
                            "save_charts": True,
                            "save_charts_path": user_defined_path,
                            "middlewares": [StreamlitMiddleware()]
                            })
```

But this doesn't save my charts to the "tmp_charts" folder. Instead it saves it to **exports->charts->temp_chart.png**

I'm working with Streamlit's chat interface where I need to save all outputs, including charts. Is there something wrong with my code here?

Side note: While using cache, I get the error "Resource temporarily unavailable". Any clues on this?

Thanks!

**Comment by nautics889** [comment]
Hello, @notmanas 
Thank you for reporting.
It appears to be a bug. I've created an issue with verbose description of the problem: https://github.com/gventuri/pandas-ai/issues/513
